lib/sarif/gosec_sarif.rb

```diff
@@ -5,28 +5,37 @@ class GosecSarif < BaseSarif
     def initialize(scan_report)
       super(scan_report)
       @uri = GOSEC_URI
-      begin
-        json_obj = JSON.parse(scan_report.log(''))
-        issues = json_obj['Issues']
-        errors = json_obj['Golang errors']
-        parsed_errors = []
-        errors.each do |key|
-          key[1].each do |location|
-            location['uri'] = key[0]
-            parsed_errors << parse_error(location)
-          end
+      @issues = Set.new
+      @logs = parse_scan_report!(scan_report)
+    end
+
+    def parse_scan_report!(scan_report)
+      json_obj = JSON.parse(scan_report.log(''))
+      issues = json_obj['Issues']
+      errors = json_obj['Golang errors']
+      parsed_errors = []
+      errors.each do |key|
+        key[1].each do |location|
+          location['uri'] = key[0]
+          parsed_errors << parse_error(location)
         end
-        @logs = issues.concat parsed_errors
-      rescue JSON::ParserError
-        @logs = []
       end
+      parsed_errors.compact!
+      issues.concat parsed_errors
+    rescue JSON::ParserError
+      []
```

Comment:
what do you think about a bugsnag log here too?  Any time we swallow an exception we should have some record of it.

Reply:
I've added a bugsnag log